### DISCUSSION TITLE
[Idea] a utility function that finds the corresponding store's value a snapshot is pointing to

### DISCUSSION THREAD

**Discussion by piscopancer** [pmndrs/valtio#837] [Q&A]
```ts
function AuthorView() {
  const snap = useSnapshot(state)
  return (
    <div>
      {snap.books.map((book, i) => (
        <Book key={book.id} book={state.books[i]} />
      ))}
    </div>
  )
}

function BookView({ book }) {
  // book is the proxy, so we can re-snap it + mutate it
  const snap = useSnapshot(book)
  return <div onClick={() => book.updateTitle()}>{snap.title}</div>
}
```
What about a utility function that finds the corresponding store's value a snapshot is pointing to (behind the scenes)?
Prolly could be used as follows:
```ts
const book = useStoreValue(bookSnap)
```
Have you thought about it?

**Comment by piscopancer** [comment]
Sorry for I may be not enough prosperous in memory management, but I'd really like to know if a function like this is possible. You are a big W tho ngl

**Comment by dai-shi** [comment]
> ```js
> const book = useStoreValue(bookSnap)
> ```

I like the idea!
The implementation might be tricky, but seems like worth exploring.
It's better to start a new project in Valtio Labs: https://github.com/valtiojs
Would you like to lead on it, or you want someone to work on it?

**Reply by piscopancer** [reply]
Honestly, I would have Already done this by myself if I knew the guts of valtio. I would have probably even done my own implementation of valtio 🤣. But at the moment, I feel like it's definitely not my business. A more responsible programmer should do it, eg you. Ofc as soon as they figure out the way to make it a thing. I just can think of an idea.

**Reply by dai-shi** [reply]
Sure, that makes sense.

I will do it when I get time, but let's see if there's any volunteer.
